### PR TITLE
Cleanup `brave_shields` component

### PR DESCRIPTION
**Pull request by cuba**
Resolves https://github.com/brave/brave-browser/issues/35837
This doesn't change any functionality simply moves files according to our guidelines so they can be used by iOS without adding more `if` dependency checks.

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by supermassive — APPROVED** [review]
wallet core lgtm

**Review by boocmp — APPROVED** [review]
tor is ok

File: browser/brave_shields/sources.gni [hunk 1]
```diff
@@ -21,8 +21,9 @@ brave_browser_brave_shields_deps = [
   "//brave/browser:browser_process",
   "//brave/common",
   "//brave/components/brave_perf_predictor/browser",
-  "//brave/components/brave_shields/browser",
-  "//brave/components/brave_shields/common",
+  "//brave/components/brave_shields/content/browser",
+  "//brave/components/brave_shields/content/test:test_support",
```
Comment (goodov) [inline browser/brave_shields/sources.gni:25]: this shouldn't be here.

File: components/brave_shields/content/test/BUILD.gn [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2024 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import("//testing/test.gni")
+
+source_set("test_support") {
```
Comment (goodov) [inline components/brave_shields/content/test/BUILD.gn:8]: `testonly = true`

File: components/brave_shields/core/browser/BUILD.gn [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2024 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+source_set("browser") {
```
Comment (goodov) [inline components/brave_shields/core/browser/BUILD.gn:6]: `static_library`

**Review by goodov — CHANGES_REQUESTED** [review]

**Review by SergeyZhukovsky — APPROVED** [review]
android browser tests ++

**Review by AlexeyBarabash — APPROVED** [review]
sync ++

File: components/brave_shields/core/browser/BUILD.gn (hunk 3 above)
Comment (cuba) [reply components/brave_shields/core/browser/BUILD.gn]: Oops I must have copied from somewhere

File: browser/brave_shields/sources.gni (hunk 1 above)
Comment (cuba) [reply browser/brave_shields/sources.gni]: I don't know why I added this. For some reason I thought it's needed here. Pushed fix.

File: components/brave_shields/content/test/BUILD.gn (hunk 2 above)
Comment (cuba) [reply components/brave_shields/content/test/BUILD.gn:8]: Pushed fix

**Review by kylehickinson — APPROVED** [review]
iOS header changes++

**Review by goodov — APPROVED** [review]

File: components/brave_shields/core/browser/DEPS [hunk 4]
```diff
@@ -0,0 +1,2 @@
+include_rules = [
```
Comment (iefremov) [inline components/brave_shields/core/browser/DEPS:1]: why is this one empty?

**Review by iefremov — APPROVED** [review]

**Review by antonok-edm — APPROVED** [review]

File: components/brave_shields/core/browser/DEPS (hunk 4 above)
Comment (cuba) [reply components/brave_shields/core/browser/DEPS]: Honestly I added this thinking I'll add something to it. But never needed to. I just removed it.

**Comment by mkarolin** [comment]
Windows CI has crashing upstream tests which appear to be flaky and aren't related to the PR changes:
```
3 tests crashed:
     PersistentBackground/ExtensionWebRequestApiTestWithContextType.WebRequestRedirectsWorkers/BackgroundResourceFetchEnabled (..\..\chrome\browser\extensions\api\web_request\web_request_apitest.cc:1099)
     ServiceWorker/ExtensionWebRequestApiTestWithContextType.WebRequestRedirectsWorkers/BackgroundResourceFetchDisabled (..\..\chrome\browser\extensions\api\web_request\web_request_apitest.cc:1099)
     ServiceWorker/ExtensionWebRequestApiTestWithContextType.WebRequestRedirectsWorkers/BackgroundResourceFetchEnabled (..\..\chrome\browser\extensions\api\web_request\web_request_apitest.cc:1099)
```

File: components/brave_shields/content/test/BUILD.gn (hunk 2 above)
Comment (bridiver) [reply components/brave_shields/content/test/BUILD.gn:8]: This is also not a valid directory within `content`. This should be inside `content/browser`

**Pull request merged**